compiler-core/src/type_/error.rs

```diff
@@ -219,6 +220,115 @@ pub enum Error {
     },
 }
 
+impl Error {
+    pub fn to_diagnostic(&self, src: String, path: PathBuf) -> Diagnostic {
```

Comment:
I think this is duplicating the existing to diagnostic code? We already convert all the type errors to diagnostics in the error module

Reply:
you're right, I've missed that.